test/writer/test_writer.py

```diff
@@ -1,27 +1,53 @@
 import numpy as np
 
 from opensquirrel import CircuitBuilder
-from opensquirrel.ir import BlochSphereRotation, ControlledGate, Float, MatrixGate, Qubit
+from opensquirrel.ir import Bit, BlochSphereRotation, ControlledGate, Float, MatrixGate, Qubit
 from opensquirrel.writer import writer
 
 
-def test_write() -> None:
+def test_circuit_without_bits() -> None:
     builder = CircuitBuilder(3)
     circuit = builder.to_circuit()
+    assert (
+        writer.circuit_to_string(circuit)
+        == """version 3.0
 
+qubit[3] q
+"""
+    )
+
+
+def test_circuit_with_qubits_and_bits() -> None:
```

Comment:
I don't see where the bits are being tested in this method. Maybe a better name would be `test_circuit_with_control_operations()`

Reply:
Very well spotted! Fixed now.